exec/java-exec/src/main/java/org/apache/drill/exec/expr/fn/impl/MappifyUtility.java

```diff
@@ -38,22 +38,29 @@
   public static final String fieldValue = "value";
 
   public static DrillBuf mappify(FieldReader reader, BaseWriter.ComplexWriter writer, DrillBuf buffer, String caller) {
-    // Currently we expect single map as input
-    if (DataMode.REPEATED == reader.getType().getMode() || !(reader.getType().getMinorType() == TypeProtos.MinorType.MAP)) {
+    // Currently we expect single map or null as input
+    if (reader.getType().getMode() == DataMode.REPEATED
+        || (reader.isSet() && reader.getType().getMinorType() != TypeProtos.MinorType.MAP)) {
       throw new DrillRuntimeException("kvgen function only supports Simple maps as input");
     }
     BaseWriter.ListWriter listWriter = writer.rootAsList();
     listWriter.startList();
     BaseWriter.MapWriter mapWriter = listWriter.map();
 
+    if (!reader.isSet()) {
```

Comment:
We can init `BaseWriter.MapWriter mapWriter = listWriter.map();` after null check.